### ISSUE TITLE
Teknoparrot 1269 - nothing is working

### ISSUE THREAD

**Issue by skolomir** [teknogods/teknoparrot-issues#238]
Please fill in the following information when creating a new issue:

**TeknoParrot version**:
1.0.0.1269

**Did this issue happen with an older version**:
Probbably, but older games have different version of teknoparrot.dll and openparrot.dll files

**Game with issue**:
Everything what I tried has problem. Watch the video on google drive link.

Private google drive link:
https://drive.google.com/file/d/1QNn_T8-DGVZxMymLvhnZKQccIiZacGDb/view?usp=sharing

**Detailed description of issue**:

Here's what specifically happens in the latest version:

For the Rambo game, after filling the screen with white dots, the game returns to the openparrot ui.

For Arcana heart 3 ex board, I get the error Failed to load dll!! (Error 4)

This game works without problems in version 1040. The suggestion to copy openparrot.dll from the old version to the new one is not the best solution. I tried that too, but such a solution is semi-usable.

HOTD Scarlet down can be opened in the latest version of teknoparrot (1269), but the controls don't work. In teknoparrot and lancer, it is possible to choose one of two exe files to launch this game - "hodzero-win64- shipping.exe" or "amdaemon.exe". With the first, the controls do not work, with the second I get an error - process state advertise. Otherwise

1. Installed directx_Jun2010_redist.exe for the official Microsoft website.
2. Nvidia drivers updated one week before.
3. All versions of Visual C++ are installed (2005 -2022).
4. Windows 10 was installed 2 months ago.
5. Windows 10 is updated.
6. Avast antivirus is turned off and folder Teknoparrot add to exceptions.
7. I also tried to run game with w7 compatibility and as administrator, didn't help too.
8. Teknoparrot is installed on C, the folder name is Teknoparrot (one word).

I have been using Windows since 1995,  it is about 30 yeras, I installed many programs, but never I had situation that nothing is working. With Teknoparrot 1269 nothing is playable - ot it is error, or controls don't work, or I'm returned to Teknoparrot UI.

Another strange thing that there  are no posts from other people with this issue. I don't know why they don't have this problem. Or they use thirdy applicatons to run games (jsconfig), or they use old/older version of Teknoparrot, or they simply don't have this problem.

I have impression that i need something more/extra to install, but don't now what. Except Teknoparrot, C++ and old direct X, I didn't find that I need some additional software.

If anyone thinks they can help me, I would like to have confirmation for the latest version of teknoparrot (1269), for the titles listed - Arcana Heart 3 Ex board, Rambo, HOTD - Scarlet down. I am interested in whether the same works in this version.

It is also possible to share the screen through the AnyDesk program.

**Reproduction details**:
Try any fame which I tried and you will see does it work.

**Any other details**:
I used old version 1040 befere few years, but I used third party application ro run games. Few of them was possible to run directly from teknoparrot ui. I coppied openparrot.dll from old version to new one and tried ro run Arcana Heart Ex board (it works in old version), but in new version controls don't work and I can't change size screen. It is never in full screen (last version), only in windowed mode. Trying to run Rombo with all dll files wasn't successfull. So, i decided to use only new version. 

I5 10 400 (6 physical/12 logical cores)
Nvidia 1650 4gb
16 GB ram
2TB SSD
Windows 10 Professional
Version: 22H2 19045.4529

**Pictures of issue**:

![01 - error code 4](https://github.com/teknogods/teknoparrot-issues/assets/69475723/32aa5d0f-9130-4cc2-9d06-6de495b2eaee)
![02 - HOTD - Scarlet down - amdaemon error](https://github.com/teknogods/teknoparrot-issues/assets/69475723/8305a495-08b8-4dcc-b249-9a0888213dbd)
![03 - file version](https://github.com/teknogods/teknoparrot-issues/assets/69475723/c041424c-70e6-4b07-8901-50651aa40364)
![04 - VC ++ - installed versions](https://github.com/teknogods/teknoparrot-issues/assets/69475723/8d42c1f3-d165-4a6e-a514-fee4a9eaaefe)

**Comment by Yoda-XY** [comment]
dude install or update Direct X and Visual C++ All-in-one thats it ;)

**Comment by skolomir** [comment]
Yes, I did fresh installation of VC++ and installed directx_Jun2010_redist.exe from Microsoft site. Today is Teknoparrot updated and still same - dll error for Arcana heart 3 exboard, no working contols for HOTD Scarlet down, no loading elf files for Rambo (after filling white dots I was return to Teknoparrot UI)..

**Comment by LordsOfTheRealm** [comment]
If anyone has this issue .you need to install Direct X 9.0 web setup. TY and Have a nice day.)
You can get it also from any SteamUnlocked Game in the _Redist folder

**Comment by M1L0S-R5** [comment]
I have installed all mentioned components but no avail - all games return Failed to load dll!! (Error 4)

Only cure is to use teknoparrot.dll and openparrot.dll files from legacy edition but this is not a solution as there are many compatibility issues (e.g. Sega Rally 3 does not recognize controls etc)

I have reinstalled Windows 11 lately and all is up to date. However, something is still missing and I cannot figure what :(

**Comment by skolomir** [comment]
I had problem with installed K-lite codec pack. When I installed it, everyhing worked. Of course, you need direct x probbably 9, VS C++, probbably all versions. I testet Terminator Salvation with latesto TP version 1.0.0.1943 and it worked. I understand your problem, because no one did not help me. Instead of empty talk, I can try something real to do with you, saomething like share screen, with TeamViewer or AnyDesk. For AnyDesk, when you download it and run, go to window install. So, tri with unistalling codesc, dowload freshh VS C++, fitst uninstall your previous versions, update graphic drivers for such case. We can speek via Viber, you can cotanct me on mail skoomir@yahoo.com for mobile number. First do whatever you can alon, than contact me. Probbably is something stupid. I have Nvidia 1650, I don't know which graphic card you have, but whatever it is, I don't think that older or newest drivers are problem. To avoid guessing, you can send me on mail name of game which you are trying to run, I will testi it on my comp first. I guess you tried more gamen, not only one. And one things more, I have W10, not W11. Try to ask someone with w11 and same grpahic card, AMD or NVIDIAD (not model, only manufacurer) to check does it work, but you already know, it works on my Nvidia 1650. Last options is to ask some AI model, Chat GPT or Claude, to see will they solve your problem.

**Comment by M1L0S-R5** [comment]
I have found a solution for this issue!

I have gaming laptop with 2 GPU's, integrated and nVidia one. All I had do is to set TeknoParrotUi.exe to be loaded by nVidia GPU from nVidia Controll Panel

Now all games work with newest TeknoParrot

**Comment by LordsOfTheRealm** [comment]
> I have found a solution for this issue!
> 
> I have gaming laptop with 2 GPU's, integrated and nVidia one. All I had do is to set TeknoParrotUi.exe to be loaded by nVidia GPU from nVidia Controll Panel
> 
> Now all games work with newest TeknoParrot


Spot on! That’s usually the fix for most laptops since they stick to the default onboard video until you manually switch it over. Great job getting that sorted

**Comment by LordsOfTheRealm** [comment]
> I have installed all mentioned components but no avail - all games return Failed to load dll!! (Error 4)
> 
> Only cure is to use teknoparrot.dll and openparrot.dll files from legacy edition but this is not a solution as there are many compatibility issues (e.g. Sega Rally 3 does not recognize controls etc)
> 
> I have reinstalled Windows 11 lately and all is up to date. However, something is still missing and I cannot figure what :(

Good luck with win 11. it blocks most of the old gamming codes EXAMPLE: dsound.dll win11 blocks it.
Won't register it and more.

**Comment by M1L0S-R5** [comment]
Well, this error took a lot of time from me.

My troubleshooting steps were:

Installed .Net 8 and TeknoParrot using Web installer
Start any game - error
Installed all Visual C++ Runtimes - Error
Installed DirectX 9 Web Installer - Error
Installed DirectX 2010 - Error
Windows 11 up to date - error
Installed LaunchBox (as someone mentioned that it helped) with some additional .Net components - Error
Installed latest full version of K-Lite Codec Pack (as someone mentioned that it helped) - Error

Finally, I have opened nVidia Controll Panel and set TeknoParrotUi.exe to be loaded by nVidia GPU - Success!

Interestingly enough, I have set up my laptop to use only nVidia GPU but it has not helped. Only when I did last step, error was gone.

I am not sure if all steps were really necessary but it does not matter, games are working now.

**Comment by LordsOfTheRealm** [comment]
as long as 
visual-c-redistributable-runtime-package-all-in-one
directx_Jun2010_redist    (2 main ones for TP to work right.)
 just for the heck of it. Installed DirectX 9 Web Installer (some older games need it)
Yes , Always do this.
NVidia control .  set TeknoParrotUi.exe to be loaded by NVidia GPU.  Otherwise it's uses the onboard Graphics
Or in Nvidia Control panel 

<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/82f6e86c-ba0b-41be-b993-97d888622050" />

**Comment by MLE010** [comment]
Same problem here on W11.. tried all of the above.
Do i really need to downgrade to W10?

Any info from developer would be great.
